src/containers/collectibles/solCollectibleHelpers.ts

```diff
@@ -0,0 +1,188 @@
+      // videoType = defaultVideoType
+    } else {
+      return null
+    }
+    return {
+      collectibleMediaType: CollectibleMediaType.VIDEO,
+      url,
+      frameUrl: nft.image || null
+    }
+  }
+  return null
+}
+
+const nftImage = async (nft: SolanaNFT): Promise<Nullable<SolanaNFTMedia>> => {
+  const files = nft.properties.files
+  // should we restrict image file extensions here?
+  // PNG, JPG, GIF
+  // https://github.com/metaplex-foundation/metaplex/blob/81023eb3e52c31b605e1dcf2eb1e7425153600cd/js/packages/web/src/views/artCreate/index.tsx#L316
+  const imageFile = files?.find(
+    file => typeof file === 'object' && file.type.includes('image')
+  ) as SolanaNFTPropertiesFile
+  const isImage =
+    nft.properties.category === 'image' || nft.image.length || imageFile
+  if (isImage) {
+    let url
+    if (nft.image.length) {
+      url = nft.image
+    } else if (imageFile) {
+      url = imageFile.uri
+    } else if (files?.length) {
+      if (files.length === 1) {
+        url = typeof files[0] === 'object' ? files[0].uri : files[0]
+      } else {
+        url = typeof files[1] === 'object' ? files[1].uri : files[1]
+      }
+    } else {
+      return null
+    }
+    return {
+      collectibleMediaType: CollectibleMediaType.IMAGE,
+      url,
+      frameUrl: url
+    }
+  }
+  return null
+}
+
+const nftComputedMedia = async (
+  nft: SolanaNFT
+): Promise<Nullable<SolanaNFTMedia>> => {
+  const files = nft.properties.files
+  if (!files?.length) {
+    return null
+  }
+  const url = typeof files[0] === 'object' ? files[0].uri : files[0]
+  // get mime type
+  // make sure it's gif/video/image
+  const headResponse = await fetch(url, { method: 'HEAD' })
+  const contentType = headResponse.headers.get('Content-Type')
+  const isGif = contentType?.includes('gif')
+  const isVideo = contentType?.includes('video')
+  if (isGif) {
+    const frameUrl = await getFrameFromGif(url, nft.name)
+    return { collectibleMediaType: CollectibleMediaType.GIF, url, frameUrl }
+  }
+  if (isVideo) {
+    return {
+      collectibleMediaType: CollectibleMediaType.VIDEO,
+      url,
+      frameUrl: null
+    }
+  }
+  return {
+    collectibleMediaType: CollectibleMediaType.IMAGE,
+    url,
+    frameUrl: url
+  }
+}
+
+export const solanaNFTToCollectible = async (
+  nft: SolanaNFT,
+  address: string
+): Promise<Collectible> => {
+  const identifier = [
+    nft.symbol,
+    nft.name,
+    nft.image /* this would not always be image e.g. could be video or gif?? */
+  ]
+    .filter(Boolean)
+    .join(':::')
+
+  const collectible = {
+    id: identifier,
+    tokenId: identifier,
+    name: nft.name,
+    description: nft.description,
+    externalLink: nft.external_url,
+    isOwned: true,
+    chain: Chain.Sol
+  } as Collectible
+
+  if (nft.properties.creators.some(creator => creator.address === address)) {
+    collectible.isOwned = false
+  }
+
+  const { url, frameUrl, collectibleMediaType } = ((await nftGif(nft)) ||
```

Comment:
Are NFTs => Collectibles really 1:1? It looks like each NFT can have multiple files, but we only get one URL/Collectible from each NFT?

Reply:
They're supposed to be 1:1 from my understanding. Each NFT can have multiple files in the metadata, but what I saw was that it's because it may be a video nft where 1 file is the video, another is the audio, another may be a thumbnail for the video...